frontend/public/src/components/forms/new_ProfileFormFields.js

```diff
@@ -25,7 +26,17 @@ export const NAME_REGEX_FAIL_MESSAGE =
   "Name cannot start with a special character (~!@&)(+:'.?,-), and cannot contain any of (/^$#*=[]`%_;\\<>{}\"|)"
 
 export const legalAddressValidation = yup.object().shape({
-  name:          yup.string().required().label("Full Name").min(2).max(254),
+  name:     yup.string().required().label("Full Name").min(2).max(254),
+  username: yup
+    .string()
+    .required()
+    .label("Public Username")
+    .min(3, usernameFieldErrorMessage)
+    .max(30, usernameFieldErrorMessage)
+    .matches(
+      /^[^!@#$%^&*()+=[]{};':"\\|,<>\/?~]*$/,
```

Comment:
This specifies a character class that contains `[]` - the closing square bracket isn't escaped, though, so the regex fails.